node/tools/src/k8s.rs

```diff
@@ -12,15 +12,168 @@ use k8s_openapi::{
 };
 use kube::{
     api::{ListParams, PostParams},
-    core::{ObjectList, ObjectMeta},
-    Api, Client, ResourceExt,
+    core::ObjectMeta,
+    Api, Client,
 };
-use std::collections::{BTreeMap, HashMap};
+use std::collections::BTreeMap;
+use std::net::SocketAddr;
 use tokio_retry::strategy::FixedInterval;
 use tokio_retry::Retry;
 use tracing::log::info;
+use zksync_consensus_roles::node;
 use zksync_protobuf::serde::Serde;
 
+/// Consensus Node Representation
+#[derive(Debug)]
+pub struct ConsensusNode {
+    /// Node identifier
+    pub id: String,
+    /// Node key
+    pub key: node::SecretKey,
+    /// Full NodeAddr
+    pub node_addr: Option<NodeAddr>,
+    /// Is seed node (meaning it has no gossipStaticOutbound configuration)
+    pub is_seed: bool,
+    /// known gossipStaticOutbound peers
+    pub gossip_static_outbound: Vec<NodeAddr>,
+}
+
+impl ConsensusNode {
+    /// Wait for a deployed consensus node to be ready and have an IP address
+    pub async fn await_running_pod(
+        &mut self,
+        client: &Client,
+        namespace: &str,
+    ) -> anyhow::Result<Pod> {
+        let pods: Api<Pod> = Api::namespaced(client.clone(), namespace);
+        // Wait until the pod is running, otherwise we get 500 error.
+        let pod = Retry::spawn(FixedInterval::from_millis(1000).take(15), || {
+            get_running_pod(&pods, &self.id)
```

Comment:
that tooling looks like an overkill for a simple retry loop with constant delay.